spmd/compiler/fusion.py

```diff
@@ -144,15 +193,18 @@ def _scan_graph_for_fusion_elements(
                 # need to fully populate this fe...
                 # we will be removing/rewriting the node list so we save prev and next
                 fe.prev_node = curr_node_list[0].prev
-                fe.next_node = node.next
+                # fe.next_node = node.next
```

Comment:
why is this commented out?

Reply:
Originally this was holding the 'next' out of comm section node (i.e we hold a prev node which is the last node before the start of comms, and a next that is the start of the regular graph (after comms).  However, next is not a reliable marker as we started doing fusions since that next node will either a) become irrelevant since this comm call was removed as part of being fused later or b)the fusion itself inserts the buffer scatter nodes here meaning that the whole comm section has expanded.  
I replaced that static pointer with the dynamic function  _get_next_node() for FusionElements. 
I have gone ahead and remove the above line so it's clear we no longer use it.